src/lib/device-online-state.ts

```diff
@@ -0,0 +1,317 @@
+import * as Bluebird from 'bluebird';
+import { EventEmitter } from 'events';
+import { DEFAULT_SUPERVISOR_POLL_INTERVAL } from './env-vars';
+import { noop } from 'lodash';
+import { resinApi, root } from '../platform';
+import { captureException } from '../platform/errors';
+import {
+	createPromisifedRedisClient,
+	PromisifedRedisClient,
+} from './redis-promise';
+import * as RedisSMQ from 'rsmq';
+import {
+	REDIS_HOST,
+	REDIS_PORT,
+	API_HEARTBEAT_STATE_TIMEOUT_SECONDS,
+} from './config';
+
+export const getPollIntervalForDevice = (uuid: string): Bluebird<number> => {
+	return resinApi
+		.get({
+			resource: 'device_config_variable',
+			passthrough: {
+				req: root,
+			},
+			options: {
+				$select: 'value',
+				$expand: {
+					device: {
+						$filter: { uuid },
+					},
+				},
+				$filter: {
+					name: 'RESIN_SUPERVISOR_POLL_INTERVAL',
+				},
+			},
+		})
+		.then(([pollInterval]: Array<{ value: number }>) => {
+			if (pollInterval) {
+				return pollInterval.value;
+			}
+
+			return resinApi
+				.get({
+					resource: 'application_config_variable',
+					passthrough: {
+						req: root,
+					},
+					options: {
+						$select: 'value',
+						$top: 1,
+						$filter: {
+							application: {
+								$any: {
+									$alias: 'a',
+									$expr: {
+										a: {
+											owns__device: {
+												$any: {
+													$alias: 'd',
+													$expr: {
+														d: {
+															uuid: uuid,
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+							name: 'RESIN_SUPERVISOR_POLL_INTERVAL',
```

Comment:
This can also be [named as `RESIN_SUPERVISOR_POLL_INTERVAL`](https://github.com/balena-io/open-balena-api/blob/master/src/lib/env-vars.ts#L31) -- unfortunately I don't think we normalise to a single prefix yet, the [PR that will do this](https://github.com/balena-io/open-balena-api/pull/34) is blocked on translations for a while now. You will need to check for both values, like so:

```
name: {
  $in: ['RESIN_...', 'BALENA_...'],
}
```

Reply:
Good spot mate, totally slipped my mind to cover that but you're absolutely right.